openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIResponseMapper.java

```diff
@@ -478,6 +480,19 @@ private void setCacheConfigField(CacheConfigAnnotation cacheConfig, String field
             case OPTIONAL_TYPE_DESC:
                 return getAPIResponses(operationAdaptor, apiResponses,
                         ((OptionalTypeDescriptorNode) typeNode).typeDescriptor(), customMediaPrefix, headers);
+            case MAP_TYPE_DESC:
+                apiResponse = setCacheHeader(headers, apiResponse, HTTP_200);
```

Comment:
Shouldn't we ideally implement these type of functionalities as setter instead of util functions? ( i.e. `apiResponse.setCacheHeader()` will be a better implementation from the OOP point of view)

Reply:
`apiResponse` object is created using the `ApiResponse` which is implemented using the swagger model. since it is a dependency model implemented one, we won't able to define a specific `setCacheHeader()` function. 🙁

Reply:
In that case maybe we can implement proxy design pattern on the swagger data model to write our own setter functions :) but since that refactoring effort will take some time, lets keep them this way for now